### PR TITLE
refactor(package_size): Move code to a python invoke task

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Refactor the `package.compare_size` method:
- move all the code on the python task (it was previously spread between `yaml` and `python`
- run all package size computations in sequential (I tried parallelism but as a resource was shared for editing it was complex for a very low gain)
- run both architecture comparison on a single job.
- use a S3 file for comparison with the common ancestor from the `main` branch instead of comparing with the last stable version of the agent.

### Motivation
https://datadoghq.atlassian.net/browse/ACIX-489

### Describe how to test/QA your changes
I added several UT to validate all the changes
Besides, I created a new task (`packages.check-size`) to enable it in parallel with the old one and assess its behaviour with less impact on users. This task [passed](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/728402642) on the last run.

### Possible Drawbacks / Trade-offs
I initially wanted to do the comparisons in parallel. But we needed to update the `package_sizes` dictionary during the multiprocess run, and this is not straightforward: you need to use a [dict manager](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.managers.SyncManager) and conversion towards a "normal" dict is not easy. As this job is not critical I dropped the use of parallelism.

### Additional Notes
Sorry in advance I did several changes in a single commit.

### CODE REVIEW

File: .gitlab/pkg_metrics/pkg_metrics.yml [hunk 1]
```diff
@@ -179,3 +179,31 @@ check_pkg_size-arm64-a7:
           ["datadog-iot-agent"]="10000000"
           ["datadog-dogstatsd"]="10000000"
       )
+
+new_check_pkg_size:
+  stage: pkg_metrics
+  image: registry.ddbuild.io/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
```
Comment (CelianR) [inline .gitlab/pkg_metrics/pkg_metrics.yml:186]: NIT: It is better to use an arm runner

File: tasks/libs/package/size.py [hunk 2]
```diff
@@ -132,3 +156,69 @@ def compute_package_size_metrics(
...
+    stable_package_size_mb = previous_size / mb
+    threshold_mb = threshold / mb
+    diff_mb = diff / mb
+    message = f"""{flavor}-{arch}-{os_name} size increase is OK:
+  New package size is {new_package_size_mb:.2f}MB
+  Ancestor package ({ancestor}) size is {stable_package_size_mb:.2f}MB
+  Diff is {diff_mb:.2f}MB (max allowed diff: {threshold_mb:.2f}MB)"""
+
+    if diff > threshold:
+        print(color_message(message.replace('OK', 'too large'), Color.RED))
```
Comment (CelianR) [inline tasks/libs/package/size.py:193]: NIT: Use stderr

**Review by CelianR — APPROVED** [review]

File: .gitlab/pkg_metrics/pkg_metrics.yml (hunk 1 above)
Comment (chouetz) [reply .gitlab/pkg_metrics/pkg_metrics.yml:186]: You're so right!

**Pull request merged**